pkg/kube/kube_handler.go

```diff
@@ -775,6 +759,10 @@ func getOCIAppRepositoryTag(cli httpclient.Client, repoURL string, repoName stri
 	}
 	defer resp.Body.Close()
 
+	if resp.StatusCode != http.StatusOK {
+		return "", fmt.Errorf("Unexpected status code when querying %q: %d", repoName, resp.StatusCode)
+	}
+
```

Comment:
This was one change which the new tests spotted as a requirement: if a non-OK status code is returned, we should not continue on and in particular, should not assume the body is parsable json.